Services/StockDeviceModels.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Exceptions;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Runtime;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.Services
+{
+    public interface IStockDeviceModels
+    {
+        IEnumerable<DeviceModel> GetList();
+        DeviceModel Get(string id);
+    }
+
+    public class StockDeviceModels : IStockDeviceModels
+    {
+        // ID used for custom device models, where the list of sensors is provided by the user
+        public const string CUSTOM_DEVICE_MODEL_ID = "custom";
+
+        private const string EXT = ".json";
+
+        private readonly IServicesConfig config;
+        private readonly ILogger log;
+
+        private List<string> deviceModelFiles;
+        private List<DeviceModel> deviceModels;
+
+        public StockDeviceModels(
+            IServicesConfig config,
+            ILogger logger)
+        {
+            this.config = config;
+            this.log = logger;
+            this.deviceModelFiles = null;
+            this.deviceModels = null;
+        }
+
+        public IEnumerable<DeviceModel> GetList()
+        {
+            if (this.deviceModels != null) return this.deviceModels;
+
+            this.deviceModels = new List<DeviceModel>();
+
+            try
+            {
+                var files = this.GetDeviceModelFiles();
+                foreach (var f in files)
+                {
+                    var c = JsonConvert.DeserializeObject<DeviceModel>(File.ReadAllText(f));
+                    c.Type = "StockModel";
+                    this.deviceModels.Add(c);
+                }
+            }
+            catch (Exception e)
+            {
+                this.log.Error("Unable to load Device Model configuration",
+                    () => new { e.Message, Exception = e });
+
+                throw new InvalidConfigurationException("Unable to load Device Model configuration: " + e.Message, e);
+            }
+
+            return this.deviceModels;
+        }
+
+        public DeviceModel Get(string id)
+        {
+            var list = this.GetList();
+            var item = list.FirstOrDefault(i => i.Id == id);
+            if (item != null)
+                return item;
+
+            this.log.Warn("Device model not found", () => new { id });
+
+            throw new ResourceNotFoundException();
```

Comment:
please add exception description, like the log msg above, including the ID

Reply:
Done.